### PR TITLE
feature/file upload

### PR DESCRIPTION
**Pull request by FRobaglia**
Closes [SVC-4815](https://zoov-eu.atlassian.net/browse/SVC-4815)

- feat: add upload icon
- feat: FFileUpload component

### CODE REVIEW

File: components/form/FFileUpload.vue [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+function isValidFile(value: unknown): boolean {
+  const isValidFormat = mimes(value, props.allowedMimeTypes);
+  const isValidSize = sizeRule(value, { size: props.maximumSize });
+
+  const validFile = isValidFormat && isValidSize;
+  if (validFile) {
+    emit('files', value as File[]);
+  }
+
+  return validFile;
```
Comment (qnp) [inline components/form/FFileUpload.vue:222]: If you use `is` before booleans, for consistency you should keep it. As you are in the scope of the funtion, the local `validFile` can be name `isValid` what do you think ?
```suggestion
  const isValid = isValidFormat && isValidSize;
  if (isValid) emit('files', value as File[]);
  return isValid;
```

File: components/FButton.vue [hunk 2]
```diff
@@ -451,6 +452,16 @@ const emit = defineEmits<{
...
+const buttonRef = ref<HTMLButtonElement>();
+
+defineExpose<{
+  ref: Ref<HTMLButtonElement | undefined>;
+  focus: () => void;
+}>({
+  ref: buttonRef,
+  focus,
+});
+
```
Comment (qnp) [inline components/FButton.vue:464]: - where is defined `focus` ?
- if it was defined, why expose `ref` instead to handle the focus of the button outside ?
- if you still need `ref`, keep it `buttonRef` for clarity

File: components/form/FFileUpload.vue [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+const iconColor = computed(() =>
+  props.disabled ? 'neutral--dark-1' : props.textColor
+);
+
+defineExpose<{
+  focus: () => void;
+}>({
+  focus,
+});
```
Comment (qnp) [inline components/form/FFileUpload.vue:247]: i would place all the "define" macros in the same region of the file, so first the `const props = withDefaults(defineProps<FFileUploadProps>()`, then the `defineEmit` then the `defineExpose`

File: stories/FFileUpload.stories.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+</div>`,
+});
+export const FocusProgrammatically: Story<FFileUploadProps> =
+  FocusTemplate.bind({});
+  
+FocusProgrammatically.args = {
+  label: 'Your student card',
+  buttonText: 'Import a file',
+  errorMessage: 'Invalid file',
+};
```
Comment (qnp) [inline stories/FFileUpload.stories.ts:102]: I tried, but i feel it's broken

**Review by qnp — CHANGES_REQUESTED** [review]

File: components/form/FFileUpload.vue (hunk 1 above)
Comment (FRobaglia) [reply components/form/FFileUpload.vue:222]: Even if it works, isn't it slightly misleading to have the same variable name in a file ?
I was thinking `isValidFormatAndSize` even though its a bit long and needs to be updated the day we add a 3rd rule .. 🤔

File: stories/FFileUpload.stories.ts (hunk 4 above)
Comment (FRobaglia) [reply stories/FFileUpload.stories.ts:102]: I believe it has to do with the fact the 2 buttons have the same `ref`. I'll investigate 🕵️

File: components/form/FFileUpload.vue (hunk 1 above)
Comment (qnp) [reply components/form/FFileUpload.vue:222]: Good point, go for `isValidFormatAndSize` 👍

File: components/form/FFileUpload.vue [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+</style>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import FButton from '@/components/FButton.vue';
+import FIcon from '@/components/FIcon.vue';
+import FField from '@/components/form/FField.vue';
+import { useInputEventBindings } from '@/composables/useInputEventBindings';
+import { useFieldWithValidation } from '@/composables/useFieldWithValidation';
+import { mimes, size as sizeRule } from '@vee-validate/rules';
```
[the remark this answers is not in the document]
Comment (qnp) [reply components/form/FFileUpload.vue:71]: Ok I see, but here it also adds contexts on a lib wich exports stuff that are not explicit so it is good to add IMO ;-)

File: components/FButton.vue [hunk 6]
```diff
@@ -334,6 +334,7 @@ import { ref, computed } from 'vue';
 import { colorDesignTokens } from '@/config/colors';
 import { getCssColor } from '@/utils/getCssColor';
 import { useFBreakpoints } from '@/composables/useFBreakpoints';
+import type { Ref } from 'vue';
```
Comment (qnp) [inline components/FButton.vue:337]: used ?

**Review by qnp — APPROVED** [review]

**Pull request merged**